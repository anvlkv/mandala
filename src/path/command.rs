use crate::{Angle, Point, Vector};

#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
#[derive(Debug, Clone, PartialEq)]
pub enum PathCommand {
    /// draw to Point
    To(PathCommandOp<Point>),
    /// draw by Vector
    By(PathCommandOp<Vector>),
}

#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
#[derive(Debug, Clone, PartialEq)]
pub enum PathCommandOp<Pv> {
    Move(Pv),
    Line(Pv),
    CubicCurve {
        to: Pv,
        ctrl1: Pv,
        ctrl2: Pv,
    },
    QudraticCurve {
        to: Pv,
        ctrl: Pv,
    },
    Arc {
        to: Pv,
        radii: Vector,
        x_rotation: Angle,
        large_arc: bool,
        sweep: bool,
    },
    ClosePath,
}
