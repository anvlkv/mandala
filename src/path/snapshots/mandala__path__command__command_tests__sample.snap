---
source: src/path/command.rs
assertion_line: 994
expression: "vec![arc_cmd.sample(0.5, from), quadratic_curve_cmd.sample(0.5, from),\n    cubic_curve_cmd.sample(0.5, from), line_cmd.sample(0.5, from),]"
---
[
    (
        -20.62942739242982,
        53.63431706922085,
    ),
    (
        50.0,
        55.0,
    ),
    (
        38.75,
        46.25,
    ),
    (
        50.0,
        50.0,
    ),
]
